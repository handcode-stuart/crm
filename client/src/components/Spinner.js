import React from "react";

import FullPageSpinner from "./styles/Spinner";

const Spinner = ({ fullpage }) => {
    return fullpage ? (
        <FullPageSpinner>
            <p>Loading...</p>
        </FullPageSpinner>
    ) : (
        <div>
            <p>Loading...</p>
        </div>
    );
};

export default Spinner;
