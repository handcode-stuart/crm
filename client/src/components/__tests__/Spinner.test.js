import React from "react";
import { shallow } from "enzyme";
import Spinner from "../Spinner";

describe("<Spinner />", () => {
    it("renders", () => {
        shallow(<Spinner />);
    });

    it("renders and matches the snapshot", () => {
        const wrapper = shallow(<Spinner />);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("renders with fullpage prop and matches the snapshot", () => {
        const wrapper = shallow(<Spinner fullpage />);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
