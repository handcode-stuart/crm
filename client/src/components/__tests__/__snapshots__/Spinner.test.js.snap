// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders and matches the snapshot 1`] = `"<div><p>Loading...</p></div>"`;

exports[`<Spinner /> renders with fullpage prop and matches the snapshot 1`] = `"<div class=\\"sc-bdVaJa hAYzll\\"><p>Loading...</p></div>"`;
